amazon-kinesis-client-multilang/src/test/java/software/amazon/kinesis/multilang/config/KinesisClientLibConfiguratorTest.java

```diff
@@ -306,12 +306,33 @@ public void testWithMissingWorkerId() {
     }
 
     @Test
-    public void testWithMissingStreamName() {
+    public void testWithMissingStreamNameAndMissingStreamArn() {
         thrown.expect(NullPointerException.class);
-        thrown.expectMessage("Stream name is required");
+        thrown.expectMessage("Stream name or Stream Arn is required. Stream Arn takes precedence if both are passed in.");
```

Comment:
**Validating exception messages is a bad smell**, precisely because of this: changing a message should not break a unit test. Validate the thrown type, not the message.

This is a variant of http://wiki.c2.com/?StringlyTyped